packages/app-spec-squad-chess/src/rules.js

```diff
@@ -39,28 +42,62 @@ const MECHANICS = {
     for(let i = 0; i < params.steps; i++) {
       to = [to[0]+params.offset[0], to[1]+params.offset[1]]
       if (!(to in position)) { break } // if off board
-      if (position[to] === null) { continue } // if empty
-      if (position[from].player === position[to].player) { break } // if same color piece
+      if (position[to].content === null) { continue } // if empty
+      if (position[from].content.player === position[to].content.player) { break } // if same color piece
       turns.push({
-        'from': from,
-        'to': to
+        from,
+        to,
+        pieceId: position[from].content.pieceId
       })
       break
     }
     return turns
   }
 }
 
-let PIECES = {}
+/* Admechanics : Mechanics as Adverbs : Verbs
+ * Mechanics generate offsets (a piece moving from one square to another)
+ * Admechanics *modify* offsets
+ */
+const ADMECHANICS = {
+  'promotion': (params, turns, position, turnNumber) => {
```

Comment:
Turn data signature should be objects with the same signature as position, which can be `Object.assigned` to the position to create a new position.